### ISSUE TITLE
\OCA\Files\Helper::getFiles() $filter does not work properly in master

### ISSUE THREAD

**Issue by plaes** [owncloud/core#20899]
This seems to be regression caused sometime during last ~3 weeks.

I have an app that looks up list of files (from external mounted OwnCloud instance) and prints them out, and is using following snippet:

```
$dir = '/External/Media';
$filter = 'httpd/unix-directory';
foreach(\OCA\Files\Helper::getFiles($dir, 'name', false, $filter) as $d) {
  // do something...
}
```
Now, after pulling latest master (8d218bf3ef842d76c2b97a175b28e13054497952), I'm not getting any directories after this call, but I managed to get things working with following workaround:

```
$dir = '/External/Media';
$filter = 'httpd/unix-directory';
foreach(\OCA\Files\Helper::getFiles($dir, 'name', false, Null) as $d) {
  if ($d->getMimeType() !== $filter)
    continue;
  // do something...
}
```

**Comment by DeepDiver1975** [comment]
@icewind1991 mind taking a look? THX

**Comment by icewind1991** [comment]
Fix is here: https://github.com/owncloud/core/pull/20904

**Comment by MorrisJobke** [comment]
Do we really support this use case? There is no public API in use and we should have this in the public API if it is really needed.

**Comment by plaes** [comment]
Yeah, it's kind of my fault for using internal API, but eventually after upgrade to 9.x I can move to the (currently WIP) JavaScript DAV.

**Comment by PVince81** [comment]
@plaes the Javascript DAV is merged on 9.0 but doesn't support mimetype filter.
You should rather use `\OC::$server->getUserFolder()->get($path)->searchByMime($mimeType)`
